libs/analytics/src/lib/components/map/map.component.ts

```diff
@@ -21,11 +22,29 @@ export class AnalyticsMapComponent {
   selected: TerritoryISOA3Value;
   top: AnalyticData[] = [];
 
+  private _topCount = new BehaviorSubject(3);
+  private _data = new Subject<AnalyticData[]>();
 
-  @Input() topCount = 3;
   @Input() @boolean showLegend = false;
   @Input() @boolean horizontal = false;
+  @Input() set topCount(count: number) {
+    this._topCount.next(count);
+  }
   @Input() set data(data: AnalyticData[]) {
+    this._data.next(data);
+  }
+
+  @Input() @boolean selectable = false;
+  @Output() selection: EventEmitter<string> = new EventEmitter();
+
+  constructor() {
+    combineLatest([
+      this._data.asObservable(),
+      this._topCount.asObservable()
+    ]).subscribe(([data, topCount]) => this.getTopCountries(data, topCount));
```

Comment:
put subscription into a variable and remove with 

```suggestion
    this.sub = combineLatest([
      this._data.asObservable(),
      this._topCount.asObservable()
    ]).subscribe(([data, topCount]) => this.getTopCountries(data, topCount));
```


```
  ngOnDestroy() {
    this.sub.unsubscribe();
  }
```